"use strict";(this.webpackChunkschemas=this.webpackChunkschemas||[]).push([[801],{801:(e,t,o)=>{o.r(t),o.d(t,{default:()=>a});const a={NCBI_SRAGenomeTrakr:{fileType:"xlsx",status:"published",method:function(e){const t=new Map([["sample_name",[]],["library_ID",[]],["title",[]],["library_strategy",[]],["library_source",[]],["library_selection",[]],["library_layout",[]],["platform",[]],["instrument_model",[]],["design_description",[]],["filetype",[]],["filename",[]],["filename2",[]],["filename3",[]],["filename4",[]],["assembly",[]],["fasta_file",[]],["enrichment_kit",[]],["amplicon_PCR_primer_scheme",[]],["library_preparation_kit",[]],["amplicon_size",[]],["quality_control_method",[]],["quality_control_method_version",[]],["quality_control_determination",[]],["quality_control_issues",[]],["quality_control_details",[]],["dehosting_method",[]],["sequence_submitter_contact_email",[]],["raw_sequence_data_processing_method",[]]]),o=[[...t.keys()]],a=e.slots,s=e.getFieldNameMap(a);e.getHeaderMap(t,a,"NCBI_SRAGenomeTrakr");for(const n of e.getTrimmedData(e.hot)){const r=[];let i;for(const[o,_]of t)i=e.getMappedField(o,n,_,a,s,"; ","NCBI_SRAGenomeTrakr"),r.push(i);o.push(r)}return o}},NCBI_SRA:{fileType:"xlsx",status:"published",method:function(e){const t=new Map([["sample_name",[]],["library_ID",[]],["title",[]],["library_strategy",[]],["library_source",[]],["library_selection",[]],["library_layout",[]],["platform",[]],["instrument_model",[]],["design_description",[]],["filetype",[]],["filename",[]],["filename2",[]],["filename3",[]],["filename4",[]],["assembly",[]],["fasta_file",[]]]),o=[[...t.keys()]],a=e.slots,s=e.getFieldNameMap(a);e.getHeaderMap(t,a,"NCBI_SRA");for(const n of e.getTrimmedData(e.hot)){const r=[];let i;for(const[o,_]of t)i=e.getMappedField(o,n,_,a,s,"; ","NCBI_SRA"),r.push(i);o.push(r)}return o}},NCBI_BIOSAMPLE_SARS_COV_2_WWS:{fileType:"xlsx",status:"published",method:function(e){const t=new Map([["sample_name",[]],["sample_title",[]],["bioproject_accession",[]],["organism",[]],["collection_date",[]],["geo_loc_name",[]],["isolation_source",[]],["ww_population",[]],["ww_sample_duration",[]],["ww_sample_matrix",[]],["ww_sample_type",[]],["ww_surv_target_1",[]],["ww_surv_target_1_known_present",[]],["collected_by",[]],["purpose_of_ww_sampling",[]],["purpose_of_ww_sequencing",[]],["sequenced_by",[]],["ww_endog_control_1",[]],["ww_endog_control_1_conc",[]],["ww_endog_control_1_protocol",[]],["ww_endog_control_1_units",[]],["ww_endog_control_2",[]],["ww_endog_control_2_conc",[]],["ww_endog_control_2_protocol",[]],["ww_endog_control_2_units",[]],["ww_flow",[]],["ww_industrial_effluent_percent",[]],["ww_ph",[]],["ww_population_source",[]],["ww_pre_treatment",[]],["ww_primary_sludge_retention_time",[]],["ww_processing_protocol",[]],["ww_sample_salinity",[]],["ww_sample_site",[]],["ww_surv_jurisdiction",[]],["ww_surv_system_sample_id",[]],["ww_surv_target_1_conc",[]],["ww_surv_target_1_conc_unit",[]],["ww_surv_target_1_extract",[]],["ww_surv_target_1_extract_unit",[]],["ww_surv_target_1_gene",[]],["ww_surv_target_1_protocol",[]],["ww_surv_target_2",[]],["ww_surv_target_2_conc",[]],["ww_surv_target_2_conc_unit",[]],["ww_surv_target_2_extract",[]],["ww_surv_target_2_extract_unit",[]],["ww_surv_target_2_gene",[]],["ww_surv_target_2_known_present",[]],["ww_surv_target_2_protocol",[]],["ww_temperature",[]],["ww_total_suspended_solids",[]],["description",[]]]),o=[[...t.keys()]],a=e.slots,s=e.getFieldNameMap(a);e.getHeaderMap(t,a,"NCBI_BIOSAMPLE_SARS_COV_2_WWS");for(const n of e.getTrimmedData(e.hot)){const r=[];let i;for(const[o,_]of t)i="organism"===o?"wastewater metagenome":e.getMappedField(o,n,_,a,s,"; ","NCBI_BIOSAMPLE_SARS_COV_2_WWS"),r.push(i);o.push(r)}return o}},NCBI_BIOSAMPLE_GenomeTrakr:{fileType:"xlsx",status:"published",method:function(e){const t=new Map([["sample_name",[]],["sample_title",[]],["bioproject_accession",[]],["organism",[]],["collection_date",[]],["collection_time",[]],["geo_loc_name",[]],["isolation_source",[]],["collection_site_id",[]],["project_name",[]],["collected_by",[]],["purpose_of_ww_sampling",[]],["ww_sample_site",[]],["ww_flow",[]],["instantaneous_flow",[]],["ww_population",[]],["ww_surv_jurisdiction",[]],["ww_population_source",[]],["ww_sample_matrix",[]],["ww_sample_type",[]],["collection_volume",[]],["ww_sample_duration",[]],["ww_temperature",[]],["ww_ph",[]],["ww_industrial_effluent_percent",[]],["ww_sample_salinity",[]],["ww_total_suspended_solids",[]],["ww_surv_system_sample_id",[]],["ww_pre_treatment",[]],["ww_primary_sludge_retention_time",[]],["specimen_processing",[]],["specimen_processing_id",[]],["specimen_processing_details",[]],["ww_processing_protocol",[]],["concentration_method",[]],["extraction_method",[]],["extraction_control",[]],["ww_endog_control_1",[]],["ww_endog_control_1_conc",[]],["ww_endog_control_1_protocol",[]],["ww_endog_control_1_units",[]],["ww_endog_control_2",[]],["ww_endog_control_2_conc",[]],["ww_endog_control_2_protocol",[]],["ww_endog_control_2_units",[]],["ww_surv_target_1",[]],["ww_surv_target_1_known_present",[]],["ww_surv_target_1_protocol",[]],["ww_surv_target_1_conc",[]],["ww_surv_target_1_conc_unit",[]],["ww_surv_target_1_gene",[]],["ww_surv_target_2",[]],["ww_surv_target_2_conc",[]],["ww_surv_target_2_conc_unit",[]],["ww_surv_target_2_gene",[]],["ww_surv_target_2_known_present",[]],["purpose_of_ww_sequencing",[]],["sequenced_by",[]],["description",[]]]),o=[[...t.keys()]],a=e.slots,s=e.getFieldNameMap(a);e.getHeaderMap(t,a,"NCBI_BIOSAMPLE_GenomeTrakr");for(const n of e.getTrimmedData(e.hot)){const r=[];let i;for(const[o,_]of t)i=e.getMappedField(o,n,_,a,s,"; ","NCBI_BIOSAMPLE_GenomeTrakr"),r.push(i);o.push(r)}return o}},ENA_GSC_MIxS_WW_ERC00023:{fileType:"xlsx",status:"published",method:function(e){const t=new Map([["project name",[]],["experimental factor",[]],["ploidy",[]],["number of replicons",[]],["extrachromosomal elements",[]],["estimated size",[]],["reference for biomaterial",[]],["annotation source",[]],["sample volume or weight for DNA extraction",[]],["nucleic acid extraction",[]],["nucleic acid amplification",[]],["library size",[]],["library reads sequenced",[]],["library construction method",[]],["library vector",[]],["library screening strategy",[]],["target gene",[]],["target subfragment",[]],["pcr primers",[]],["multiplex identifiers",[]],["adapters",[]],["pcr conditions",[]],["sequencing method",[]],["sequence quality check",[]],["chimera check software",[]],["relevant electronic resources",[]],["relevant standard operating procedures",[]],["negative control type",[]],["positive control type",[]],["collection date",[]],["geographic location (country and/or sea)",[]],["geographic location (latitude)",[]],["geographic location (longitude)",[]],["geographic location (region and locality)",[]],["depth",[]],["broad-scale environmental context",[]],["local environmental context",[]],["environmental medium",[]],["source material identifiers",[]],["sample material processing",[]],["isolation and growth condition",[]],["propagation",[]],["amount or size of sample collected",[]],["oxygenation status of sample",[]],["organism count",[]],["sample storage duration",[]],["sample storage temperature",[]],["sample storage location",[]],["sample collection device",[]],["sample collection method",[]],["biochemical oxygen demand",[]],["chemical oxygen demand",[]],["pre-treatment",[]],["primary treatment",[]],["reactor type",[]],["secondary treatment",[]],["sludge retention time",[]],["tertiary treatment",[]],["host disease status",[]],["host scientific name",[]],["alkalinity",[]],["industrial effluent percent",[]],["sewage type",[]],["wastewater type",[]],["temperature",[]],["pH",[]],["efficiency percent",[]],["emulsions",[]],["gaseous substances",[]],["inorganic particles",[]],["organic particles",[]],["soluble inorganic material",[]],["soluble organic material",[]],["suspended solids",[]],["total phosphate",[]],["nitrate",[]],["phosphate",[]],["salinity",[]],["sodium",[]],["total nitrogen",[]],["subspecific genetic lineage",[]],["trophic level",[]],["relationship to oxygen",[]],["known pathogenicity",[]],["encoded traits",[]],["observed biotic relationship",[]],["chemical administration",[]],["perturbation",[]]]),o=[[...t.keys()]],a=e.slots,s=e.getFieldNameMap(a);e.getHeaderMap(t,a,"ENA_GSC_MIxS_WW_ERC00023");for(const n of e.getTrimmedData(e.hot)){const r=[];let i;for(const[o,_]of t){if("source material identifiers"===o){const e=s["specimen collector subsample ID"],t=s["specimen collector sample ID"];let o=void 0!==e?n[e]:"",a=void 0!==t?n[t]:"";i=o&&""!==o.trim()?o:a&&""!==a.trim()?a:""}else i=e.getMappedField(o,n,_,a,s,"; ","ENA_GSC_MIxS_WW_ERC00023");r.push(i)}o.push(r)}return o}},ENA_SEWAGE_ERC00036:{fileType:"xlsx",status:"published",method:function(e){const t=new Map([["name of the sampling site",[]],["nucleic acid extraction",[]],["nucleic acid amplification",[]],["investigation type",[]],["surveillance target",[]],["collection date",[]],["geographic location (country and/or sea)",[]],["geographic location (latitude)",[]],["geographic location (longitude)",[]],["geographic location (region and locality)",[]],["amount or size of sample collected",[]],["sample storage duration",[]],["sample storage temperature",[]],["sample storage location",[]],["sampling time point",[]],["sample transportation temperature",[]],["sample transportation date",[]],["sample transportation time",[]],["receipt date",[]],["sewage type",[]],["temperature",[]],["area of sampling site",[]],["size of the catchment area",[]],["population size of the catchment area",[]]]),o=[[...t.keys()]],a=e.slots,s=e.getFieldNameMap(a);e.getHeaderMap(t,a,"ENA_SEWAGE_ERC00036");for(const n of e.getTrimmedData(e.hot)){const r=[];let i;for(const[o,_]of t)i=e.getMappedField(o,n,_,a,s,"; ","ENA_SEWAGE_ERC00036"),r.push(i);o.push(r)}return o}},VirusSeq_Portal:{fileType:"tsv",status:"published",method:function(e){const t=new Map([["study_id",[]],["specimen collector sample ID",[]],["GISAID accession",[]],["sample collected by",[]],["sequence submitted by",[]],["sample collection date",[]],["sample collection date null reason",[]],["geo_loc_name (country)",[]],["geo_loc_name (state/province/territory)",[]],["organism",[]],["isolate",[]],["fasta header name",["isolate"]],["purpose of sampling",[]],["purpose of sampling details",[]],["anatomical material",[]],["anatomical part",[]],["body product",[]],["environmental material",[]],["environmental site",[]],["collection device",[]],["collection method",[]],["host (scientific name)",[]],["host disease",[]],["host age null reason",[]],["host age unit",[]],["host age bin",[]],["host gender",[]],["purpose of sequencing",[]],["purpose of sequencing details",[]],["sequencing instrument",[]],["sequencing protocol",[]],["raw sequence data processing method",[]],["dehosting method",[]],["consensus sequence software name",[]],["consensus sequence software version",[]],["breadth of coverage value",[]],["depth of coverage value",[]],["reference genome accession",[]],["bioinformatics protocol",[]],["gene name",[]]]),o=new Map([["not applicable","Not Applicable"],["missing","Missing"],["not collected","Not Collected"],["not provided","Not Provided"],["restricted access","Restricted Access"]]),a={"Alberta Precision Labs (APL)":"ABPL-AB","BCCDC Public Health Laboratory":"BCCDC-BC","Manitoba Cadham Provincial Laboratory":"MCPL-MB","New Brunswick - Vitalité Health Network":"VHN-NB","Newfoundland and Labrador - Eastern Health":"EH-NL","Nova Scotia Health Authority":"NSHA-NS","Public Health Ontario (PHO)":"PHO-ON","Laboratoire de santé publique du Québec (LSPQ)":"LSPQ-QC","Saskatchewan - Roy Romanow Provincial Laboratory (RRPL)":"RRPL-SK"},s=e.slots,n=e.getFieldNameMap(s);e.getHeaderMap(t,s,"VirusSeq_Portal");const r=[[...t.keys()]],i=new Set(["xsd:nonNegativeInteger","xsd:decimal"]);for(const l of e.getTrimmedData(e.hot)){const p=[];var _=!1;for(const[r,u]of t)if(!0===_)_=!1;else{var c=e.getMappedField(r,l,u,s,n,":","VirusSeq_Portal");if("study_id"==r){const e=l[n.sequence_submitted_by];e&&e in a&&(c=a[e])}if("breadth of coverage value"==r&&c&&c.length&&"%"!=c.substr(-1)&&(c+="%"),t.has(r+" null reason"))if(u.length>0){const e=s[n[u[0]]];e?e.dataStatus&&e.dataStatus.includes(c)?(p.push(""),_=!0):o.has(c.toLowerCase())?(c=o.get(c.toLowerCase()),p.push(""),_=!0):i.has(e.datatype)&&isNaN(Number(c))&&(p.push(""),_=!0):alert('Template configuration error: "'+r+'" has misnamed source field.')}else alert('Template configuration error: "'+r+'" has no source mapped field.');p.push(c)}r.push(p)}return r}},GISAID:{fileType:"xls",status:"published",method:function(e){const t=[["Submitter",[]],["FASTA filename",[]],["Virus name",[]],["Type",[]],["Passage details/history",[]],["Collection date",[]],["Location",[]],["Additional location information",[]],["Host",[]],["Additional host information",[]],["Sampling Strategy",[]],["Gender",[]],["Patient age",[]],["Patient status",[]],["Specimen source",[]],["Outbreak",[]],["Last vaccinated",[]],["Treatment",[]],["Sequencing technology",[]],["Assembly method",[]],["Coverage",["depth_of_coverage_value"]],["Originating lab",[]],["Address",["sample_collector_contact_address"]],["Sample ID given by the sample provider",[]],["Submitting lab",[]],["Address",["sequence_submitter_contact_address"]],["Sample ID given by the submitting laboratory",["specimen_collector_sample_id"]],["Authors",[]]],o=e.slots,a=e.getFieldNameMap(o);e.getHeaderMap(t,o,"GISAID");const s=[Array.from(t,(e=>e[0]))];for(const n of e.getTrimmedData(e.hot)){const e=[];for(const[s]of t.entries()){const r=t[s][0],i=t[s][1];if("Type"===r){e.push("betacoronavirus");continue}const _=[];for(const e of i){let t=a[e],s=n[t];if(!s)continue;const r=o[t],c=s.toLowerCase().trim();if("specimen_processing"===r.name){if(!c.split(";").includes("virus passage"))continue;s="Virus passage"}if(r.dataStatus&&r.dataStatus.map((e=>e.toLowerCase().trim())).includes(c)){if(i.length>1)continue;s="Unknown"}"passage_number"===r.name&&(s="passage number "+s),_.push(s)}"Assembly method"===r?e.push(_.join(":")):e.push(_.join(";"))}s.push(e)}return s.splice(0,0,["submitter","fn","covv_virus_name","covv_type","covv_passage","covv_collection_date","covv_location","covv_add_location","covv_host","covv_add_host_info","covv_sampling_strategy","covv_gender","covv_patient_age","covv_patient_status","covv_specimen","covv_outbreak","covv_last_vaccinated","covv_treatment","covv_seq_technology","covv_assembly_method","covv_coverage","covv_orig_lab","covv_orig_lab_addr","covv_provider_sample_id","covv_subm_lab","covv_subm_lab_addr","covv_subm_sample_id","covv_authors"]),s}}}}}]);